### PR TITLE
Some bug fix to make it runnable on sdfstudio

### PR DESCRIPTION
**Pull request by quantaji**

### CODE REVIEW

**Comment by quantaji** [comment]
@hermannsblum

File: nerfstudio/data/dataparsers/sdfstudio_dataparser.py [hunk 1]
```diff
@@ -116,17 +111,21 @@ def get_semantics(image_idx: int, semantic_path, group_path):
...
+    
     # groups for patch loss
-    groups_raw = np.array(Image.open(group_path[image_idx]), dtype="uint8")
-    groups_raw = torch.from_numpy(groups_raw).type(torch.uint8)
-    # make sure groups are unique and start from 0
-    groups = torch.zeros_like(groups_raw)
-    for i, group_id in enumerate(groups_raw.unique()):
-        groups[groups_raw == group_id] = i
-    groups = groups.type(torch.uint8)
+    if group_path[image_idx] is not None:
```
Comment (hermannsblum) [inline nerfstudio/data/dataparsers/sdfstudio_dataparser.py]: this fixes the error, but it will make it hard to debug because the rest of the code now does not know whether the data includes masks or not.  
I think I would prefer a more explicit version where you add an option to the config, e.g. "include_groups" that just defaults to false

File: nerfstudio/models/base_surface_model.py [hunk 2]
```diff
@@ -266,8 +266,16 @@ def populate_modules(self):
         self.ssim = structural_similarity_index_measure
         self.lpips = LearnedPerceptualImagePatchSimilarity()
         if hasattr(self, 'semantics'):
-            self.sem_accuracy = Accuracy(task='multiclass', num_classes=self.semantics_numclasses)
-            self.sem_miou = JaccardIndex(task='multiclass', num_classes=self.semantics_numclasses)
+            self.sem_accuracy = Accuracy(
+                task='multiclass', 
+                num_classes=self.semantics_numclasses, 
+                ignore_index=self.config.semantic_ignore_label
```
Comment (hermannsblum) [inline nerfstudio/models/base_surface_model.py:272]: good fix!

File: scripts/render.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def _render_trajectory_video(
         output_image_dir.mkdir(parents=True, exist_ok=True)
     else:
         install_checks.check_ffmpeg_installed()
-    semantics_model2output = pipeline.model.semantics_model2output.copy().cpu()
+    semantics_model2output = pipeline.model.semantics_model2output.detach().cpu()
```
Comment (hermannsblum) [inline scripts/render.py:75]: nice, did this make it faster?

**Review by hermannsblum — COMMENTED** [review]
does it make sense to you to add a config for the masks? I think it would be cleaner to configure the dataset in a certain way and then always knowing what you can expect from it, or the data loading already failing if you want masks but they are not there

File: scripts/render.py (hunk 3 above)
Comment (quantaji) [reply scripts/render.py:75]: I remember the `.copy()` raises error, therefore I change to `.detach()`

File: nerfstudio/data/dataparsers/sdfstudio_dataparser.py (hunk 1 above)
Comment (quantaji) [reply nerfstudio/data/dataparsers/sdfstudio_dataparser.py]: My latest commit switch to this approach, with a new config `include_semantic_groups`. It runs with no error.

**Review by hermannsblum — APPROVED** [review]
